src/main/scala/ai/starlake/schema/handlers/HdfsStorageHandler.scala

```diff
@@ -215,9 +215,21 @@ class HdfsStorageHandler(fileSystem: String)(implicit
     }
   }
 
+  /** Copy file
+    * @param src
+    *   source path
+    * @param dst
+    *   destination path
+    * @return
+    */
+  override def copy(src: Path, dst: Path): Boolean = {
+    val currentFS = fs(src)
+    FileUtil.copy(currentFS, src, currentFS, dst, false, conf)
```

Comment:
We would have FileUtil.copy(fs(src), src, fs(dst), dst …)

Reply:
@tiboun I think @mhdkassir that delivered  exactly what we were searching for. I will be backport this FileUtil.copy right away

Reply:
fixed

> We would have FileUtil.copy(fs(src), src, fs(dst), dst …)

Reply:
Great :) @hayssams  exactly what we wanted to implement.